File: MultiLanguageSandbox/src/test/deps/haskell/50.hs
calculateExpression :: Int -> Int -> Int -> Int
calculateExpression a b c = (a + b) * c
check :: (Int -> Int -> Int -> Int) -> IO ()
check calculateExpressionFunc = do
  let testCases = 
        [ ((2, 3, 5), 25),
          ((-1, 1, 10), 0),
          ((10, 20, -3), -90),
          ((0, 0, 100), 0),
          ((-100, 100, 0), 0)
        ]

  mapM_ (\((a, b, c), expected) -> testCase a b c expected) testCases
  where
    testCase a b c expected = do
      let result = calculateExpressionFunc a b c
      putStrLn $
        "Input: a=" ++ show a ++ ", b=" ++ show b ++ ", c=" ++ show c ++
        ", Expected: " ++ show expected ++ ", Result: " ++ show result ++
        if result == expected then " (Pass)" else error "Test failed. Exiting..."

main :: IO ()
main = check calculateExpression

File: MultiLanguageSandbox/src/test/deps/perl/50.pl

sub calculate_expression {
    my ($a, $b, $c) = @_;
    return ($a + $b) * $c;
}


sub check {
    my $func = shift;
    die "Test failed!" unless &$func(2, 3, 5) == 25;
    die "Test failed!" unless &$func(10, 20, 3) == 90;
    die "Test failed!" unless &$func(-1, 1, 1) == 0;
    die "Test failed!" unless &$func(0, 0, 1) == 0;
}

check(\&calculate_expression);